ios/LocalNetworkPrivacy/LocalNetworkPrivacy.m

```diff
@@ -0,0 +1,47 @@
+#import <UIKit/UIKit.h>
+#import "LocalNetworkPrivacy.h"
+
+@interface LocalNetworkPrivacy () <NSNetServiceDelegate>
+
+@property (nonatomic) NSNetService *service;
+@property (nonatomic) void (^completion)(BOOL);
+@property (nonatomic) NSTimer *timer;
+@property (nonatomic) BOOL publishing;
+
+@end
+
+@implementation LocalNetworkPrivacy
+
+- (instancetype)init {
+    if (self = [super init]) {
+        self.service = [[NSNetService alloc] initWithDomain:@"local." type:@"_lnp._tcp." name:@"LocalNetworkPrivacy" port:1100];
+    }
+    return self;
+}
+
+- (void)dealloc {
+    [self.service stop];
+}
+
+- (void)checkAccessState:(void (^)(BOOL))completion {
```

Comment:
Is this as reliable as:
(It seems you altered the code from: https://stackoverflow.com/a/65800221)
```
    func checkAccessState(completion: @escaping (Bool) -> Void) {
        self.completion = completion
        
        timer = .scheduledTimer(withTimeInterval: 2, repeats: true, block: { timer in
            guard UIApplication.shared.applicationState == .active else {
                return
            }
            
            if self.publishing {
                self.timer?.invalidate()
                self.completion?(false)
            }
            else {
                self.publishing = true
                self.service.delegate = self
                self.service.publish()
                
            }
        })
    }
```